src/s3.ts

```diff
@@ -641,6 +645,27 @@ export const countS3ObjectNameAppearances = async (
   return counter;
 };
 
+export async function checkDirectory(
```

Comment:
```suggestion
/**
 * This is a helper function that resolves whether a given path
 * is a directory, because the S3 API does not provide this in listings.
export async function isDirectory(
```